integration/singularity/store.go

```diff
@@ -194,18 +194,43 @@ func (l *SingularityStore) Start(ctx context.Context) error {
 
 	logger.Info("Checking %v storage providers", len(l.storageProviders))
 	for _, sp := range l.storageProviders {
-		logger.Info("Checking storage provider %s", sp)
-		var foundSchedule bool
+		var foundSchedule *models.ModelSchedule
 		logger := logger.With("provider", sp)
 		for _, schd := range listPreparationSchedulesRes.Payload {
 			scheduleAddr, err := address.NewFromString(schd.Provider)
 			if err == nil && sp == scheduleAddr {
-				foundSchedule = true
-				logger.Infow("Schedule found for provider", "id", schd.ID)
+				foundSchedule = schd
 				break
 			}
 		}
-		if !foundSchedule {
+		if foundSchedule != nil {
+			// If schedule was found, update it
+			logger.Infow("Schedule found for provider. Updating with latest settings...", "id", foundSchedule.ID)
+			l.singularityClient.DealSchedule.UpdateSchedule(&deal_schedule.UpdateScheduleParams{
```

Comment:
You should check for an error no?

Reply:
ah, good catch, thanks!

is there no lint for that?